rewards/explorer.py

```diff
@@ -11,7 +14,7 @@
 }
 
 
-def fetch_block_by_timestamp(chain: str, timestamp: int):
+def fetch_block_by_timestamp(chain: str, timestamp: int) -> Optional[Dict]:
```

Comment:
API / HTTP requests should be handled through a function which catches any non-200 responses and retries with maximum amount of retries.  If retries are exhausted, log to discord and gracefully exit.